packages/contracts/contracts/governance/MintManager.sol

```diff
@@ -0,0 +1,136 @@
+    constructor(
+        address _governanceToken,
+        address _owner,
+        address[] memory _recipients,
+        uint256[] memory _shares
+    ) {
+        GOVERNANCE_TOKEN = GovernanceToken(_governanceToken);
+
+        transferOwnership(_owner);
+
+        require(_recipients.length == _shares.length, "MintManager: invalid length of array");
+
+        uint256 totalShares = 0;
+        for (uint256 i = 0; i < _recipients.length; i++) {
+            address recipient = _recipients[i];
+            require(recipient != address(0), "MintManager: recipient address cannot be 0");
+
+            uint256 share = _shares[i];
+            require(share != 0, "MintManager: share cannot be 0");
+
+            totalShares += share;
+            recipients.push(recipient);
+            shareOf[recipient] = share;
+        }
+
+        require(
+            totalShares <= SHARE_DENOMINATOR,
+            "MintManager: max total share is equal or less than SHARE_DENOMINATOR"
+        );
+    }
+
+    /**
+     * @notice Only the owner is allowed to mint mint cap amount of the GovernanceToken at once.
+     */
+    function mint() external onlyOwner {
+        require(!minted, "MintManager: already minted on this chain");
+
+        uint256 mintCap = MINT_CAP * 10 ** GOVERNANCE_TOKEN.decimals();
+
+        uint256 totalAmount;
+        for (uint256 i = 0; i < recipients.length; i++) {
+            address recipient = recipients[i];
+            uint256 share = shareOf[recipient];
+            uint256 amount = (mintCap * share) / SHARE_DENOMINATOR;
+            totalAmount += amount;
+        }
+
+        GOVERNANCE_TOKEN.mint(address(this), totalAmount);
+
+        minted = true;
+    }
+
+    /**
+     * @notice Only the owner is allowed to distribute the GovernanceToken to specified recipients.
+     */
+    function distribute() external onlyOwner {
+        uint256 mintCap = MINT_CAP * 10 ** GOVERNANCE_TOKEN.decimals();
+
+        for (uint256 i = 0; i < recipients.length; i++) {
+            address recipient = recipients[i];
+            uint256 share = shareOf[recipient];
+            uint256 amount = (mintCap * share) / SHARE_DENOMINATOR;
+            GOVERNANCE_TOKEN.transfer(recipient, amount);
+        }
+
+        uint256 balance = GOVERNANCE_TOKEN.balanceOf(address(this));
+        require(balance == 0, "MintManager: tokens remain after distribution");
+    }
+
+    /**
+     * @notice Only the owner is allowed to renounce the ownership of the GovernanceToken.
+     */
+    function renounceOwnershipOfToken() external onlyOwner {
+        GOVERNANCE_TOKEN.renounceOwnership();
+    }
+
+    /**
+     * @notice Only the owner is allowed to transfer the ownership of the GovernanceToken.
+     *
+     * @param newMintManager The new MintManager to own the GovernanceToken.
+     */
+    function transferOwnershipOfToken(address newMintManager) external onlyOwner {
+        GOVERNANCE_TOKEN.transferOwnership(newMintManager);
+    }
+}
```

Comment:
Is this function required?

Reply:
I added this function to prevent the worst case if the owner of governance token contract should be changed to new `MintManager` due to a bug. It was added because `GovernanceToken` was a predeployed contract, but now we can just use new proxy contract in that worst case. Although need to find new proxy address using new salt, do you think that it's better to just remove this function?